### PR TITLE
feature #2517 and #2518 - assets wallet main area and mock transactions

### PR DESCRIPTION
**Pull request by goranjovic**
addresses #2517 and #2518 and also #2516 

### Summary:

UI for the main area of Assets screen (balance and change) and transactions history list.

Transactions history list uses the same data binding as the Main wallet transactions list. Filtering based on asset type (e.g. only SNT tokens on SNT asset screen) will be handled in a separate PR

These changes are hidden behind a flag and do not affect the publicly accessible parts of the app.

status: ready

### CODE REVIEW

**Comment by oskarth** [comment]
@goranjovic To clarify, is this in progress or ready for review? PR says status: ready and it seems to have been moved back from review immediately

File: src/status_im/ui/screens/wallet/assets/subs.cljs
Comment (jeluard) [inline src/status_im/ui/screens/wallet/assets/subs.cljs]: Use `assoc-in` ?

File: src/status_im/ui/screens/wallet/main/views.cljs
Comment (jeluard) [inline src/status_im/ui/screens/wallet/main/views.cljs]: When need a better way to pass data to screens..

**Review by jeluard — APPROVED** [review]

**Comment by goranjovic** [comment]
Sorry @oskarth I missed your comment. It was still in progress until now, we decided to merge another PR first shortly after I requested review.

It is ready to review now, though.

File: src/status_im/ui/screens/wallet/main/views.cljs
Comment (goranjovic) [reply src/status_im/ui/screens/wallet/main/views.cljs]: That we definitely do. I have an idea around that, but need to research the navigation support first.

File: src/status_im/ui/screens/wallet/assets/styles.cljs
Comment (oskarth) [inline src/status_im/ui/screens/wallet/assets/styles.cljs]: align

File: src/status_im/ui/screens/wallet/assets/views.cljs
Comment (oskarth) [inline src/status_im/ui/screens/wallet/assets/views.cljs:22]: Hm, if we can avoid shadowing a core fn (symbol) that'd be good IMO. Is there something wrong with calling it `token-symbol`? Aside from the repetition. Or maybe there's a different name we can use?

File: src/status_im/ui/screens/wallet/assets/views.cljs
Comment (oskarth) [inline src/status_im/ui/screens/wallet/assets/views.cljs:36]: Not 100% sure but I think token->unit deals with amount being nil? So might not need the (or amount 0)

File: src/status_im/ui/screens/wallet/assets/views.cljs
Comment (oskarth) [inline src/status_im/ui/screens/wallet/assets/views.cljs:80]: Same re shadowing core fns

File: src/status_im/ui/screens/wallet/main/views.cljs
Comment (janherich) [reply src/status_im/ui/screens/wallet/main/views.cljs]: Please have a look at the `status-im.chat.event` namespace and how the navigation to chat screen combined with preselecting data works there, there is really no need for this double dispatch. 
We should avoid using the `:set`/`:set-in` event like a plaque (same goes for equivalent subscriptions, hopefully, it will be removed soon) and really not introduce any new code doing this.

File: src/status_im/ui/screens/wallet/main/views.cljs
Comment (oskarth) [inline src/status_im/ui/screens/wallet/main/views.cljs:3]: re-frame?

**Review by janherich — CHANGES_REQUESTED** [review]

File: src/status_im/utils/ethereum/tokens.cljs
Comment (oskarth) [inline src/status_im/utils/ethereum/tokens.cljs:8]: I guess the shadowing name symbol stuff already existed so maybe this is outside of scope of PR. It should work with lexical scope regardless, so maybe it's fine in small functions. Not sure.

**Comment by goranjovic** [comment]
@janherich There would've been no need for data preselection either if only our navigation supported parameters. That's the source of the problem and that is where we should try to fix this without introduction of additional complexity.

**Comment by jeluard** [comment]
@janherich @goranjovic That is something I plan to look at soon as part of the performance improvements. Our whole navigation stack is old and slugish. One potential replacement could be https://reactnavigation.org/

File: src/status_im/ui/screens/wallet/assets/views.cljs
Comment (goranjovic) [reply src/status_im/ui/screens/wallet/assets/views.cljs:22]: It was `token-symbol` in the mocked code. This rename is to conform with the map with all token info. In general, I agree with this - too bad all good (i.e generic var names) are already taken by core.

**Comment by janherich** [comment]
@goranjovic What additional complexity ? Just params for navigation won't cut it, since sometimes not only data-preselection is needed, but also additional action like firing-off some side-effect (`web3`/`jail` whatever).
There is no problem whatsoever doing that just combining the functions for navigation (manipulating the navigation stack) + business logic attached to navigation as a plain old function composition, returning `re-frame` effects map responsible for everything, in just one atomic action hooked for example under event named `navigate-to-wallet-token`

**Comment by goranjovic** [comment]
@jeluard When I took a look into this, it seemed to me that the only thing that prevents us from passing params to `navigate-to` right now is the fact that there is a "browsing history" stack. That said, there's nothing to prevent us from adding params along with screen identifiers to that stack. That said, we should try to use something existing rather than the homemade one we have for a bunch of reasons.

**Comment by goranjovic** [comment]
@janherich A whole lot of times it really is just a parameter and I do think it is a kludgy to create a special event for every single instance, even if there are cases when only that will work.
The analogy here is say a redirection to a web page with a parameter, say `/users/63` you just do that, redirect to a page... and params.

**Comment by janherich** [comment]
@goranjovic You can still do that without additional event dispatch even with our current navigation, just overloading multi-method `preload-data!` for keyword representing the screen you want navigate to, even if I don't think it's a elegant approach, because the logic of what is being preloaded is kind of "detached".
Regarding the second thing, when you don't need to do anything besides manipulating the navigation stack, fine, just fire off `:navigate-to` with proper screen-id. Whenever you need to have some special logic attached to it (other effect besides swapping the nav stack in app-db), wrap-it with special event for that purpose, so it's not every single instance. 
So please, if you don't wan't do go with the special event, at least use the `preload-data!` multimethod to get rid of that `:set` event :)

**Comment by goranjovic** [comment]
@janherich Event wrapper seems like a reasonably elegant solution, working on it. Do you mind if we moved this PR forward and merge the refactor in the next one (the one I'm working on right now) in order to unblock @jeluard as he has to rebase against my UI changes?

**Comment by goranjovic** [comment]
@janherich Actually, ignore the previous message, I wrapped it in an event, as you suggested. Are you happy with it? I'd really like to merge this one, as it is blocking @jeluard who needs to rebase against my UI changes.

**Review by janherich — APPROVED** [review]

**Pull request merged**
